scalding-core/src/main/scala/com/twitter/scalding/Config.scala

```diff
@@ -110,6 +110,29 @@ trait Config {
   def setMapSideAggregationThreshold(count: Int): Config =
     this + (AggregateBy.AGGREGATE_BY_THRESHOLD -> count.toString)
 
+  /**
+   * This function gets the set of classes that have been registered to Kryo.
+   * They may or may not be used in this job, but Cascading might want to be made aware
+   * that these classes exist
+   */
+  def getKryoRegisteredClasses: Set[Class[_]] = {
+    // Get an instance of the Kryo serializer (which is populated with registrations)
+    getKryo.map { kryo =>
+      val cr = kryo.newKryo.getClassResolver
+
+      @annotation.tailrec
+      def kryoClasses(idx: Int, acc: Set[Class[_]]): Set[Class[_]] =
+        Option(cr.getRegistration(idx)) match {
+          case present @ Some(reg) => kryoClasses(idx + 1, acc ++ present.map { _.getType }
+            .filterNot { _.isPrimitive } // Cascading handles primitives and arrays
```

Comment:
Any preference on the top options? keep public and move filters, or make private and leave as is, maybe changing the comment/function name to match the fact its filtered?